### ISSUE TITLE
Generic.ControlStructures.DisallowYodaConditions false positives

### ISSUE THREAD

**Issue by AlexHowansky** [squizlabs/PHP_CodeSniffer#2897]
The Generic.ControlStructures.DisallowYodaConditions sniff incorrectly triggers when the left side of a condition includes a function or static method call inside an empty() call. Tested with phpcs 3.5.4 and PHP 7.4.3.

Does not trigger [correct]:
```
if (empty($a) === true) {
    exit;
}
```
Triggers [incorrect]:
```
if (empty(a()) === true) {
    exit;
}
```
Does not trigger [correct]:
```
if (empty($a->a()) === true) {
    exit;
}
```
Triggers [incorrect]:
```
if (empty(a::a()) === true) {
    exit;
}
```
